askama_derive/src/generator.rs

```diff
@@ -237,18 +236,52 @@ fn find_used_templates(
                 _ => {}
             }
         }
-        map.insert(path, source);
+        map.insert(path, parsed);
     }
     Ok(())
 }
 
+mod _parsed {
+    use std::mem;
+
+    use crate::config::Syntax;
+    use crate::parser::{parse, Node};
+    use crate::CompileError;
+
+    pub(super) struct Parsed {
+        #[allow(dead_code)]
+        source: String,
+        nodes: Vec<Node<'static>>,
+    }
+
+    impl Parsed {
+        pub(super) fn new(source: String, syntax: &Syntax<'_>) -> Result<Self, CompileError> {
+            // Self-referential borrowing: `self` will keep the source alive as `String`,
+            // internally we will transmute it to `&'config str` to satisfy the compiler.
+            // However, we only expose the nodes with a lifetime limited to `self`.
+            let src = unsafe { mem::transmute::<_, &'static str>(source.as_str()) };
```

Comment:
Maybe explicitly use `mem::transmute::<&str, &'static str>`, so we can be sure that we are not changing the type, but only its lifetime?